### ISSUE TITLE
TS1109 when using an IHtmlHelper extension within a Razor View

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#85593]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/TS1109-when-using-an-IHtmlHelper-extensi/10855884)._

---
[severity:It bothers me. A fix would be nice]
Create a new ASP.NET Core WebApp (MVC)

Define an IHtmlHelper extension like this:

```
		public static IHtmlContent SomeExtension(this IHtmlHelper html)
		{
			return html. Raw("content");
		}
```

Use it within a cshtml view like this:

```
	&lt;script&gt;

var content = @Html.SomeExtension();

&lt;/script&gt;
```

Getting this warning:

![TS1109.png](https://aka.ms/dc/image?name=B58c4d5492a7e44889fbade9b464affbc638757837386834827_TS1109.png&tid=58c4d5492a7e44889fbade9b464affbc638757837386834827)

If I put the statement within quotes, the warning goes away. However, the purpose of the extension is to preserve the type of the content. So for example if the output value of the extension is "true", then the javascript line will become "var content = true;" which means that the type of "content" will be a boolean which is the desired outcome. If I put it in quotes then it will be a string.

The code still works, the only issue is the warning at this point which started coming up after an update to VS2022. Not sure which one, I believe it was around late 17.10 or early 17.11.

My current version is **17.12.5**

Thank you!

---
#### Feedback Bot on 2/23/2025, 11:32 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 2/27/2025, 02:47 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>
